NExtends/Primitives/DateTimes/DateTime.extensions.cs

```diff
@@ -194,5 +198,98 @@ public static DateTime AddDaysSafe(this DateTime date, double value)
 				return DateTime.MaxValue;
 			}
 		}
+
+        public static string GetDayOrdinalSuffix(this DateTime date, CultureInfo culture)
+        {
+            switch (culture.LCID)
+            {
+                case 1033:
+                case 2057:
+                    return GetEnglishDayOrdinalSuffix(date);
+                case 1036:
+                    return GetFrenchDayOrdinalSuffix(date);
+                case 1031:
+                    return GetGermanDayOrdinalSuffix(date);
+                case 2067:
+                    return GetDutchDayOrdinalSuffix(date);
+                default:
+                    return "";
+            }
+        }
+
+        public static DateTime GetLastSundayOfYearISO8601(int year)
+        {
+            var lastDayOfYear = new DateTime(year, 12, 31);
+            if (lastDayOfYear.DayOfWeek == DayOfWeek.Sunday)
+                return lastDayOfYear;
+
+            var gapDaysNumberToSunday = (lastDayOfYear.DayOfWeek >= DayOfWeek.Thursday ? 7 - (int)lastDayOfYear.DayOfWeek : -(int)lastDayOfYear.DayOfWeek);
+            return lastDayOfYear.AddDays(gapDaysNumberToSunday);
+        }
+
+        public static DateTime GetStartOfWeek(this DateTime value)
+        {
+            var date = value.Date;
+            //DayOfWeek starts on Sunday but we want weeks starting on monday
+            int daysIntoWeek = (int)date.DayOfWeek == 0 ? 6 : (int)date.DayOfWeek - 1;
+            return date.AddDays(-daysIntoWeek);
+        }
+
+        public static string ToUserCultureFormat(this DateTime date)
+        {
+            return date.ToString(System.Threading.Thread.CurrentThread.CurrentCulture.DateTimeFormat.ShortDatePattern);
+        }
+
+        // http://stackoverflow.com/a/20175
+        private static string GetEnglishDayOrdinalSuffix(DateTime date)
+        {
+            switch (date.Day % 100)
+            {
+                case 11:
+                case 12:
+                case 13:
+                    return "th";
+            }
+
+            switch (date.Day % 10)
+            {
+                case 1:
+                    return "st";
+                case 2:
+                    return "nd";
+                case 3:
+                    return "rd";
+                default:
+                    return "th";
+            }
+        }
+
+        private static string GetFrenchDayOrdinalSuffix(DateTime date)
+        {
+            if (date.Day == 1)
+            {
+                return "er";
+            }
+            else
+            {
+                return "";
+            }
+        }
+
+        private static string GetGermanDayOrdinalSuffix(DateTime date)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter 'date'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1172)